plugin.php

```diff
@@ -35,39 +37,91 @@ function queue_export( $config = null ) {
  * @param  string $config
  */
 function static_page_save( $config = null ) {
-
 	// Track the background task in an option for reporting / status checking.
 	$update_progress = [
 		'date'      => time(),
 		'urls'      => [],
 		'done_urls' => [],
 	];
+
 	$option_name = 'static_page_save_running';
 	update_option( $option_name, $update_progress );
 
-	$urls = get_site_urls();
+	$posts_per_page = 50;
 
-	$update_progress['urls'] = $urls;
-	update_option( $option_name, $update_progress );
+	$query_args = [
+		'post_type'              => 'any',
+		'posts_per_page'         => $posts_per_page,
+		'paged'                  => 1,
+		'fields'                 => 'ids',
+		'no_found_rows'          => true,
+		'update_post_meta_cache' => false,
+		'update_post_term_cache' => false,
+	];
+
+	$query = new WP_Query( $query_args );
+	if ( ! $query->have_posts() ) {
+		return;
+	}
+
+	$total_pages = $query->max_num_pages;
```

Comment:
Newb question: IF we set `paged` to `1` in the query, will this still return the maximum number of pages (ie: more than 1 page)?